pkg/resourcemanager/webhook/highavailabilityconfig/handler.go

```diff
@@ -159,6 +159,7 @@ func (h *Handler) handleDeployment(
 		maxReplicas,
 		&deployment.Spec.Template,
 		deployment.Annotations,
+		metav1.LabelSelector{MatchLabels: deployment.Spec.Template.Labels},
```

Comment:
Do we also want to handle Deployments in a similar way in the future? If yes, we could add a TODO so that we don't forget it.
I see that they are not affected as the etcd StatefulSet but from the PR description I get the impression that we didn't do it now to do not roll everything at the same time but we still want to do it in the future.

Reply:
I'm actually uncertain about this point. Technically we don't need it and would only do so for the sake of consistency. So we would probably wait for the "right" time, i.e. when a lot of deployments are updates anyway. But will we really consider or see this TODO then? That being said, I think the TODO will rather be a tombstone.